providers/supermicro/supermicrox/supermicrox.go

```diff
@@ -247,7 +248,7 @@ func (s *SupermicroX) HardwareType() (model string) {
 	m, err := s.Model()
 	if err != nil {
 		// Here is your sin
-		s.log.V(1).Info("error getting hardwaretype", "err", err.Error())
+		s.log.V(1).Info("error getting hardwaretype", "err", internal.ErrStringOrEmpty(err))
```

Comment:
probably only need the `ErrStringOrEmpty` for loglines that check `if err != nil || ...`. other than that looks good, thanks!

Reply:
I changed it everywhere to be more consistent but I can change it only for `if err != nil || ...` if you think it is better that way.

Reply:
up to you. it's not needed in some places, but I don't think it hurts either.